crates/muvm/src/server/bin/muvm-server.rs

```diff
@@ -1,31 +1,58 @@
+use std::env;
 use std::os::unix::process::ExitStatusExt as _;
+use std::path::PathBuf;
+use std::str::FromStr;
 
-use anyhow::Result;
+use anyhow::{Context, Result};
 use log::error;
 use muvm::server::cli_options::options;
 use muvm::server::worker::{State, Worker};
+use nix::unistd::geteuid;
 use tokio::net::TcpListener;
 use tokio::process::Command;
 use tokio::sync::watch;
 use tokio_stream::wrappers::WatchStream;
 use tokio_stream::StreamExt as _;
+use uuid::Uuid;
 
-#[tokio::main]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
+    let cookie = env::var("MUVM_SERVER_COOKIE")
+        .with_context(|| "Could find server cookie as an environment variable")?;
+
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    rt.block_on(async { tokio_main(cookie).await })
+}
+
+async fn tokio_main(cookie: String) -> Result<()> {
     env_logger::init();
 
-    let options = options().run();
+    let cookie = Uuid::from_str(&cookie).context("Couldn't parse cookie as UUID v7")?;
```

Comment:
[`Uuid::from_str`](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.from_str) just calls [`Uuid::parse_str`](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.parse_str), and the docs also mentions to use [`Uuid::try_parse`](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.try_parse) unless we need detailed error information, which I don't think we do in this case?

Reply:
You're right, `try_parse` it's better. Changed.